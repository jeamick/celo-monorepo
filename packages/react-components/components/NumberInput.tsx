import colors from '@celo/react-components/styles/colors'
import * as React from 'react'
import { StyleSheet, TextInput, TextInputProperties, View, ViewStyle } from 'react-native'

interface Props {
  onChange: (pin1: string) => void
  onSubmit: () => void
  value?: string
  style?: ViewStyle
  isSensitiveInput?: boolean
  keyboardType?: TextInputProperties['keyboardType']
  textContentType?: any
  placeholder?: string
  autoFocus?: boolean
  testID?: string
}

export default class NumberInput extends React.Component<Props> {
  render() {
    const {
      style: propsStyle,
      onChange,
      onSubmit,
      value = '',
      isSensitiveInput = false,
      keyboardType = 'default',
      textContentType = 'none',
      placeholder = '',
      autoFocus = false,
      testID,
    } = this.props

    return (
      <View style={[style.container, propsStyle]}>
        <TextInput
          style={[style.borderedText, style.numberInput]}
          onChangeText={onChange}
          value={value}
          underlineColorAndroid="transparent"
          placeholder={placeholder}
          keyboardType={keyboardType}
          secureTextEntry={isSensitiveInput}
          // @ts-ignore until we upgrade '@types/react-native'
          textContentType={textContentType}
          maxLength={6}
          onSubmitEditing={onSubmit}
          autoFocus={autoFocus}
          testID={testID}
        />
      </View>
    )
  }
}

const style = StyleSheet.create({
  container: {
    paddingHorizontal: 60,
    paddingVertical: 15,
  },
  borderedText: {
    borderWidth: 1,
    borderColor: colors.inputBorder,
    borderRadius: 3,
    padding: 7,
    fontSize: 24,
  },
  numberInput: {
    textAlign: 'center',
    backgroundColor: '#FFFFFF',
  },
})
