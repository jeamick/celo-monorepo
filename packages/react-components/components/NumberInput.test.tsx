import NumberInput from '@celo/react-components/components/NumberInput'
import { shallow } from 'enzyme'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

describe('NumberInput', () => {
  it('renders correctly with minimum props', () => {
    const tree = renderer.create(<NumberInput onSubmit={jest.fn()} onChange={jest.fn()} />)
    expect(tree).toMatchSnapshot()
  })
  it('delegates', () => {
    const numberInput = shallow(
      <NumberInput
        onSubmit={jest.fn()}
        onChange={jest.fn()}
        testID="Decentralize!"
        value="1"
        isSensitiveInput={true}
      />
    )
    expect(numberInput.find({ testID: 'Decentralize!' }).prop('value')).toEqual('1')
    expect(numberInput.find({ testID: 'Decentralize!' }).prop('secureTextEntry')).toEqual(true)
  })
})
