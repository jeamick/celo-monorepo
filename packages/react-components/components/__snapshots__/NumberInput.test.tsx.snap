// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput renders correctly with minimum props 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 60,
        "paddingVertical": 15,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoFocus={false}
    keyboardType="default"
    maxLength={6}
    onChangeText={[MockFunction]}
    onSubmitEditing={[MockFunction]}
    placeholder=""
    rejectResponderTermination={true}
    secureTextEntry={false}
    style={
      Array [
        Object {
          "borderColor": "#D1D5D8",
          "borderRadius": 3,
          "borderWidth": 1,
          "fontSize": 24,
          "padding": 7,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "textAlign": "center",
        },
      ]
    }
    textContentType="none"
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
